near/omni-bridge/src/lib.rs

```diff
@@ -489,18 +496,89 @@ impl Contract {
         );
 
         let mut required_storage_balance =
-            self.add_transfer_message(transfer_message.clone(), storage_payer.clone());
+            self.required_balance_for_init_transfer_message(transfer_message.clone());
         required_storage_balance = required_storage_balance.saturating_add(
             NearToken::from_yoctonear(init_transfer_msg.native_token_fee.0),
         );
 
+        let message_storage_account_id =
+            Self::calculate_message_storage_account_id(&transfer_message, &signer_id);
+        if self.has_storage_balance(&message_storage_account_id, required_storage_balance) {
+            self.init_transfer_internal(transfer_message, message_storage_account_id, signer_id);
+        } else if self.has_storage_balance(&signer_id, required_storage_balance) {
+            self.init_transfer_internal(transfer_message, signer_id.clone(), signer_id);
+        } else {
+            let promise_index = env::promise_yield_create(
+                "init_transfer_resume",
+                json!({
+                    "transfer_message": transfer_message,
+                    "message_storage_account_id": message_storage_account_id,
+                    "storage_owner": signer_id,
+                })
+                .to_string()
+                .as_bytes(),
+                INIT_TRANSFER_RESUME_GAS,
+                GasWeight(0),
+                PROMISE_REGISTER_ID,
+            );
+
+            let yield_id: CryptoHash = env::read_register(PROMISE_REGISTER_ID)
+                .expect("ERR_READ_PROMISE_REGISTER")
+                .try_into()
+                .expect("ERR_READ_PROMISE_REGISTER");
+
+            self.init_transfer_promises
```

Comment:
It will be good later to figure out how to reserve a storage for this in advance, for example it could be done by decreasing the balance of self account.

Reply:
Makes sense. Done here 87ed79352aebf10129ac766f42d1ce88c74b55d5